Neos.Neos/Classes/Command/UserCommandController.php

```diff
@@ -288,27 +317,51 @@ public function addRoleCommand($username, $role, $authenticationProvider = null)
      * related to the given provider. However, once a user has been found, the role will be removed from <b>all</b>
      * existing accounts related to that user, regardless of its authentication provider.
      *
-     * @param string $username The username of the user
+     * @param string $username The username of the user (globbing is supported)
      * @param string $role Role to be removed from the user, for example "Neos.Neos:Administrator" or just "Administrator"
      * @param string $authenticationProvider Name of the authentication provider to use. Example: "Neos.Neos:Backend"
      * @return void
      */
     public function removeRoleCommand($username, $role, $authenticationProvider = null)
     {
-        $user = $this->getUserOrFail($username, $authenticationProvider);
+        $users = $this->findUsersByUsernamePattern($username, $authenticationProvider);
+        if (empty($users)) {
+            $this->outputLine('No users that match name-pattern "%s" do exist.', array($username));
+            $this->quit(1);
+        }
 
-        try {
-            if ($this->userService->removeRoleFromUser($user, $role) > 0) {
-                $this->outputLine('Removed role "%s" from user "%s".', array($role, $username));
-            } else {
-                $this->outputLine('User "%s" did not have the role "%s" assigned.', array($username, $role));
+        foreach ($users as $user) {
+            $username = $this->userService->getUsername($user, $authenticationProvider);
+            try {
+                if ($this->userService->removeRoleFromUser($user, $role) > 0) {
+                    $this->outputLine('Removed role "%s" from user "%s".', array($role, $username));
+                } else {
+                    $this->outputLine('User "%s" did not have the role "%s" assigned.', array($username, $role));
+                }
+            } catch (NoSuchRoleException $exception) {
+                $this->outputLine('The role "%s" does not exist.', array($role));
+                $this->quit(2);
             }
-        } catch (NoSuchRoleException $exception) {
-            $this->outputLine('The role "%s" does not exist.', array($role));
-            $this->quit(2);
         }
     }
 
+    /**
+     * Find all users the match the given username with globbing support
+     *
+     * @param string $usernamePattern pattern for the username of the users to find
+     * @param string $authenticationProviderName Name of the authentication provider to use
+     * @return array<User>
+     */
+    protected function findUsersByUsernamePattern($usernamePattern, $authenticationProviderName = null)
+    {
+        return array_filter(
```

Comment:
This can be pretty expensive with a lot of users. I am wary of adding features that have obvious performance problems. Especially as now even without actually using globbing this method is used. So if you have too many users it will be broken with or without globbing usage.

Reply:
I optimized this a bit to avoid fetching all users if no globbing pattern is given. On the other hand this does pretty much the same as the method `getTableRowForUser` does for all users during `user:list`.